cabal-install/src/Distribution/Client/TargetSelector.hs

```diff
@@ -467,8 +467,9 @@ resolveTargetSelectors :: KnownTargets
 resolveTargetSelectors (KnownTargets{knownPackagesAll = []}) [] _ =
     ([TargetSelectorNoTargetsInProject], [])
 
-resolveTargetSelectors (KnownTargets{knownPackagesPrimary = []}) [] _ =
-    ([TargetSelectorNoTargetsInCwd], [])
+-- if the component kind filter is just exes, we don't want to suggest "all" as a target.
+resolveTargetSelectors (KnownTargets{knownPackagesPrimary = []}) [] ckf =
+    ([TargetSelectorNoTargetsInCwd (ckf /= Just ExeKind) ], [])
```

Comment:
What about parametrize `TargetSelectorNoTargetsInCwd` with the type of ckf instead a Bool? Maybe it will give information to refine the error in front of other target types or show more info in the error

Reply:
I started doing this but I don't think it helps much because this is just for _one_ specific error message and there's really only the binary option to suggest "all" or not. We can generalize if this ever changes.